nemo_aligner/models/nlp/gpt/gpt_sft_model.py

```diff
@@ -32,6 +37,35 @@
 
 
 class GPTSFTModel(MegatronGPTModel, SupervisedInterface):
+    def __init__(self, cfg, trainer):
+        super().__init__(cfg, trainer)
+        inference_params = dict(cfg.get("inference", {}))
+        # note that this will fail is import path is not available when the model is restored
+        # this is by design as it might not be possible to use model correctly without a matching
+        # inference strategy
+        if "strategy" in inference_params:
+            inference_params["strategy"] = hydra.utils.instantiate(inference_params["strategy"], model=self)
+        self.set_inference_params(**inference_params)
+
+    def set_inference_params(self, length_params=None, sampling_params=None, strategy=None):
+        # TODO (igitman): the name self._inference_params is very similar to self.inference_params
+        #    that's used by the base model for another purpose. There is also self._inference_config
+        #    that has a similar role to the parameters below but is less convenient.
+        #    While there is a danger for accidental name collision and this adds confusion, it's ok for now
+        #    as we are planning to remove dependence on the MegatronGPTModel after which we can remove this note
+        # registering inference parameters with default values
+        self._inference_params = {
+            "length_params": get_default_length_params().copy(),
+            "sampling_params": get_default_sampling_params().copy(),
+            "strategy": strategy,
+        }
+        # overriding any non-default values from the config
+        self._inference_params["sampling_params"].update(sampling_params or {})
+        self._inference_params["length_params"].update(length_params or {})
```

Comment:
I think the code above would be easier to follow if using the more straightforward:
```python
if length_params is None:
    length_params = get_default_length_params().copy()
if sampling_params is None:
    sampling_params = get_default_sampling_params().copy()
```

Reply:
but that's not equivalent, since we are doing an update here on top of the default parameters. Or do you mean something like this?

```python
        full_length_params = get_default_length_params().copy()
        if length_params is None:
            full_length_params.update(length_params)
        full_sampling_params = get_default_sampling_params().copy()
        if sampling_params is None:
            full_sampling_params.update(sampling_params)

        self._inference_params = {
            "length_params": full_length_params,
            "sampling_params": full_sampling_params,
            "strategy": strategy,
        }
```

Reply:
Oh my bad, I didn't realize that the idea was that you could specify only a subset of the settings in the args.

Makes sense now => resolving.